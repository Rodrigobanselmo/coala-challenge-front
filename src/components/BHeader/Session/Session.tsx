import { Box, Button, Typography, styled } from "@mui/material";

import { useAuth } from "@/core/contexts/AuthContext";
import { ISessionProps } from "./types";

export function Session(props: ISessionProps): JSX.Element {
  const { user, googleSignIn, signOut } = useAuth();

  return (
    <Box display="flex">
      <Button
        variant={user ? "outlined" : "contained"}
        onClick={user ? signOut : googleSignIn}
        sx={{ mr: 10, minWidth: "100px" }}
      >
        {user ? "Sair" : "Entrar"}
      </Button>

      <Typography color="text.main" fontSize={["18px", "20px", "22px"]}>
        {user ? "Fializar Sessão" : "Iniciar Sessão"}
      </Typography>
    </Box>
  );
}
