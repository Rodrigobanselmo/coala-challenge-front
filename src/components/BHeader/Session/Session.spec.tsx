import React from "react";
import { render } from "@testing-library/react";
import { Session } from "./Session";

jest.mock("../../../core/contexts/AuthContext", () => ({
  useAuth: () => ({
    user: null,
    googleSignIn: () => {},
    signOut: () => {},
  }),
}));

describe("Session", () => {
  it("matches the snapshot when user is null", () => {
    const { container } = render(<Session />);
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot when user is authenticated", () => {
    jest.mock("../../../core/contexts/AuthContext", () => ({
      useAuth: () => ({
        user: {
          id: "1",
          email: "qwe@qwe.com",
        },
        googleSignIn: () => {},
        signOut: () => {},
      }),
    }));

    const { container } = render(<Session />);
    expect(container).toMatchSnapshot();
  });
});
