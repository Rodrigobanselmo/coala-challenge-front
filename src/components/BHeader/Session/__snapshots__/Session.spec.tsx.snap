// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session matches the snapshot when user is authenticated 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-5xoqx6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Entrar
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pezyni-MuiTypography-root"
    >
      Iniciar Sessão
    </p>
  </div>
</div>
`;

exports[`Session matches the snapshot when user is null 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-5xoqx6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Entrar
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pezyni-MuiTypography-root"
    >
      Iniciar Sessão
    </p>
  </div>
</div>
`;
